src/lib/swapper/swappers/LifiSwapper/LifiSwapper3.ts

```diff
@@ -0,0 +1,53 @@
+import type { ChainAdapter, EvmChainId } from '@shapeshiftoss/chain-adapters'
+import type { HDWallet } from '@shapeshiftoss/hdwallet-core'
+import { Err, Ok } from '@sniptt/monads'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import type { Swapper3 } from 'lib/swapper/api'
+import { makeSwapErrorRight, SwapErrorType } from 'lib/swapper/api'
+import { getUnsignedTx } from 'lib/swapper/swappers/LifiSwapper/getUnsignedTx/getUnsignedTx'
+
+import type { LifiTradeQuote } from './utils/types'
+
+export const lifi: Swapper3 = {
+  getUnsignedTx: async (tradeQuote: LifiTradeQuote<false>, wallet: HDWallet) => {
+    const { selectedLifiRoute } = tradeQuote
+    const { accountNumber, sellAsset } = tradeQuote.steps[0]
+
+    const unsignedTx = await getUnsignedTx({
+      selectedLifiRoute,
+      accountNumber,
+      sellAsset,
+      wallet,
+    })
+
+    return unsignedTx
+  },
+  executeTrade: async ({ txToExecute, wallet, chainId }) => {
```

Comment:
Logic not abstracted into its own helper like `getUnsignedTx` because there's already a `lib/swapper/swappers/LifiSwapper/executeTrade/executeTrade`.
This is effectively the same logic, but stripped down as a pure function returning a Txid without the status polling logic.

We may want to split the logic of the module above into two, so we can reuse it here

Reply:
Also now that `executeTrade` actually executes a trade (as in, signs and broadcast an unsigned Tx/a message), we should be able to abstract this into two variants and keep it DRY:
- one for signing and broadcasting Txs
- one for signing and broadcasting messages

Reply:
excellent, i think we'll need to explore how to break out our specific implementation details from the generalised pattern so we can tick off this part of the "swappers slash swap api" mega project